src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -22,42 +23,94 @@ public DefaultClaimUidExtractor(ObjectPool<AntiforgerySerializationContext> pool
         }
 
         /// <inheritdoc />
-        public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
+        public string ExtractClaimUid(ClaimsPrincipal claimsPrincipal)
         {
-            if (claimsIdentity == null || !claimsIdentity.IsAuthenticated)
+            if (claimsPrincipal == null)
+            {
+                throw new ArgumentNullException(nameof(claimsPrincipal));
+            }
+
+            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsPrincipal.Identities);
+            if (uniqueIdentifierParameters == null)
             {
                 // Skip anonymous users
                 return null;
             }
 
-            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsIdentity);
             var claimUidBytes = ComputeSha256(uniqueIdentifierParameters);
             return Convert.ToBase64String(claimUidBytes);
         }
 
-        // Internal for testing
-        internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity claimsIdentity)
+        public static IList<string> GetUniqueIdentifierParameters(IEnumerable<ClaimsIdentity> claimsIdentities)
         {
-            var nameIdentifierClaim = claimsIdentity.FindFirst(
-                claim => string.Equals(ClaimTypes.NameIdentifier, claim.Type, StringComparison.Ordinal));
-            if (nameIdentifierClaim != null && !string.IsNullOrEmpty(nameIdentifierClaim.Value))
+            foreach (var identity in claimsIdentities)
```

Comment:
Not sure but I suspect going through the `claimsIdentities` twice makes it worthwhile saving the enumerable to an array. Then again is this another place where the enumerable is always backed by a `List<T>`.